src/main/java/io/r2dbc/postgresql/codec/PointCodec.java

```diff
@@ -0,0 +1,65 @@
+package io.r2dbc.postgresql.codec;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.r2dbc.postgresql.client.Parameter;
+import io.r2dbc.postgresql.message.Format;
+import io.r2dbc.postgresql.type.PostgresqlObjectId;
+import io.r2dbc.postgresql.util.Assert;
+import io.r2dbc.postgresql.util.ByteBufUtils;
+
+import static io.r2dbc.postgresql.message.Format.FORMAT_BINARY;
+import static io.r2dbc.postgresql.type.PostgresqlObjectId.POINT;
+
+public class PointCodec extends AbstractCodec<Point> {
+
+    private final ByteBufAllocator byteBufAllocator;
+
+    PointCodec(ByteBufAllocator byteBufAllocator) {
+        super(Point.class);
+        this.byteBufAllocator = Assert.requireNonNull(byteBufAllocator, "byteBufAllocator must not be null");
+    }
+
+    @Override
+    boolean doCanDecode(PostgresqlObjectId type, Format format) {
+        Assert.requireNonNull(type, "type must not be null");
+        Assert.requireNonNull(format, "format must not be null");
+
+        return POINT == type && format == FORMAT_BINARY;
+    }
+
+    @Override
+    Point doDecode(ByteBuf buffer, PostgresqlObjectId dataType, Format format, Class<? extends Point> type) {
+        Assert.requireNonNull(buffer, "byteBuf must not be null");
+        Assert.requireNonNull(type, "type must not be null");
+        Assert.requireNonNull(format, "format must not be null");
+
+        try {
+            String decodedAsString = ByteBufUtils.decode(buffer);
+            String parenRemovedVal = decodedAsString.replaceAll("[()]", "");
+            String[] coordinatesAsString = parenRemovedVal.split(",");
+            double x = Double.parseDouble(coordinatesAsString[0]);
+            double y = Double.parseDouble(coordinatesAsString[1]);
+            return new Point(x, y);
+        } catch (NumberFormatException | NullPointerException | ArrayIndexOutOfBoundsException e) {
```

Comment:
We don't catch `NumberFormatException` in Numeric codecs and catching `NullPointerException` and `ArrayIndexOutOfBoundsException` looks kind of weird.

Reply:
Actually `NumberFormatException` was caught in [pgjdbc](https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/geometric/PGpoint.java#L75-L78), i thought, the same should be done here as well. Also, these other exceptions were also possible if we receive invalid data, hence caught these exceptions. I can remove it.

Reply:
What @Squiry said. Let's see whether we have a need for handling such cases at all.